frb_dart/lib/src/basic.dart

```diff
@@ -129,12 +126,16 @@ class FlutterRustBridgeTask<S> extends FlutterRustBridgeBaseTask {
 
 /// A task to call FFI function, but it is synchronous.
 @immutable
-class FlutterRustBridgeSyncTask extends FlutterRustBridgeBaseTask {
+class FlutterRustBridgeSyncTask<S> extends FlutterRustBridgeBaseTask {
   /// The underlying function to call FFI function, usually the generated wire function
   final WireSyncReturnStruct Function() callFfi;
 
+  /// Parse the returned data from the underlying function
+  final S Function(dynamic) parseSuccessData;
```

Comment:
Given that we always have Uint8List as the raw data, can we make the type more specific? i.e.

1. maybe here it is "Uint8List" not "dynamic"
2. In the generated code, such as `int _wire2api_SyncReturn_i64(dynamic raw) {`, we can change `dynamic raw` to `Uint8List raw`

Reply:
What do you think about
```
format!(
        "{} _wire2api_{}({}{} raw) {{
            {}
        }}
        ",
        ty.dart_api_type(),
        ty.safe_ident(),
        extra_argument,
        match ty {
            SyncReturn(_) => "Uint8List",
            _ => "dynamic",
        },
        body,
    )

Reply:
What about asking `ty` to tell what it wants, like `ty.whatever_function_name()`

Reply:
That's surely a better idea if I'm allowed to add a new API to IrType :)
What about
```
#[inline]
    pub fn dart_param_type(&self) -> &'static str {
        match self {
            SyncReturn(_) => "Uint8List",
            _ => "dynamic",
        }
    }

Reply:
Well I would prefer:

```
    pub fn dart_param_type(&self) -> &'static str {
        "dynamic" // <- this is the default impl
    }

impl IrSyncReturn {
  dart_param_type() { "Uint8List" }
}
```

Reply:
Indeed, this is better 😆
I will commit once CI passes

Reply:
CI is already passing now

![image](https://user-images.githubusercontent.com/5236035/188271100-25f1a323-3680-4039-9fcc-27b3a50f1b5c.png)

Reply:
I created a new branch to run CI before committing to this PR branch :)
https://github.com/SoLongAndThanksForAllThePizza/flutter_rust_bridge/runs/8169528268

Reply:
Good solution to avoid having me to click the "allow to run ci" over and over again :) I will add this trick to doc to tell further PRers

Reply:
https://github.com/fzyzcjy/flutter_rust_bridge/blob/master/.github/PULL_REQUEST_TEMPLATE.md

Reply:
Glad I could help 😸

Reply:
> I created a new branch to run CI before committing to this PR branch :) https://github.com/SoLongAndThanksForAllThePizza/flutter_rust_bridge/runs/8169528268

By the way, do you know how to run CI in a forked repository before pr? Does `.github/workflows/ci.yaml` need to be changed?

Reply:
I'm curious about this too